lib/pavilion/test_config/resolver.py

```diff
@@ -671,12 +671,37 @@ def resolve_permutations(test_cfg, base_var_man):
                 raise TestConfigError(
                     "Permutation variable '{}' contains index or subvar."
                     .format(per_var))
-            elif base_var_man.is_deferred(var_set, var):
+            elif base_var_man.any_deferred(per_var):
                 raise TestConfigError(
-                    "Permutation variable '{}' references a deferred variable."
+                    "Permutation variable '{}' references a deferred variable "
+                    "or one with deferred components."
                     .format(per_var))
             used_per_vars.add((var_set, var))
 
+        if test_cfg.get('subtitle', None) is None:
+            subtitle = []
+            valid_subtitle = True
+            var_dict = base_var_man.as_dict()
+            for per_var in permute_on:
+                var_set, var, index, subvar = base_var_man.resolve_key(per_var)
+                if isinstance(var_dict[var_set][var][0], dict):
+                    sub_key = list(var_dict[var_set][var][0].keys())[0]
+                    subtitle.append(''.join(['{{', per_var, '.', sub_key,
+                                             '}}']))
+                    valid_subtitle = False
+                else:
+                    subtitle.append('{{' + per_var + '}}')
+
+            subtitle = '-'.join(subtitle)
+
+            if valid_subtitle:
+                test_cfg['subtitle'] = subtitle
+            else:
+                raise TestConfigError(
```

Comment:
I guess you decided not to just raise a warning.  I'm still not sure an exception is the right way to go.  A lack of a subtitle is inconvenient, but I don't think it means the tests shouldn't run.  It just means you'll have to go into the results or the run directory to figure out which one each one is.

Reply:
We'll talk about this at the meeting today.